http.go

```diff
@@ -43,9 +43,10 @@ type ServerHandler struct {
 }
 
 func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
-	debugLog := log.WithPrefix(sh.Logger, level.Key(), level.DebugValue())
-	infoLog := log.WithPrefix(sh.Logger, level.Key(), level.InfoValue())
-	errorLog := log.WithPrefix(sh.Logger, level.Key(), level.ErrorValue())
+	logger := log.With(logging.GetLogger(request.Context()))
```

Comment:
As a defense mechanism, it might be good to fall back here on `sh.Logger` if `logging.GetLogger(...)` results in a nil logger.